app/services/cache_service.rb

```diff
@@ -43,7 +47,34 @@ def listing_unchanged?(prev_cached_listing, fresh_listing)
       (prev_cached_listing['LastModifiedDate'] == fresh_listing['LastModifiedDate'])
   end
 
+  def listing_images_unchanged?(prev_cached_listing, fresh_listing)
+    prev_cached_listing_images = prev_cached_listing.present? && prev_cached_listing['Listing_Images']
+    fresh_listing_images = fresh_listing.present? && fresh_listing['Listing_Images']
+
+    return true if fresh_listing_images.blank?
```

Comment:
Does this condition ever actually happen?

Reply:
Yes, if someone deletes the last listing image in the array, the array could be empty